### PR TITLE
JIT - Floating Point Support for Expression JIT

### PR DESCRIPTION
**Pull request by bdw**
This PR adds basic floating point support for the expression JIT.
Some of the design decisions:
- no different operators for arithmetic operations, store, copy etc. between integer and floating point operands. (Basically, every operator whose implementation and output types are determined completely by the input type, is treated as generic, because the tiler can sort out the difference perfectly). Types are checked by the expression template compiler and to a lesser extent by the JIT.
- in the register allocator (and emitter), treat register sets consistently as bitmaps, register identifiers as numbers. In x86-64, we treat the register set as a 32-bit bitmap (16 GPR, 16 SSE registers, ignoring x87 and MMX registers for the moment!). Live ranges also get a set of permissible registers - this helps to handle register requirerements.
- Register allocator now enforces 'two-operand' order - i.e, the output argument should be the same, as the first input argument, for binary operators.

### CODE REVIEW

**Comment by bdw** [comment]
Oh, it also pulls in an instruction encoding fix for DynASM.

File: src/jit/core_templates.expr [hunk 1]
```diff
@@ -171,12 +193,25 @@
...
   (call (^func &MVM_coerce_s_i)
     (arglist
       (carg (tc) ptr)
       (carg $1   ptr)) int_sz))
 
+(template: coerce_sn
+  (calln (^func &MVM_coerce_s_n)
+    (arglist
+      (carg (tc) ptr)
+      (carg $1   ptr))))
```
Comment (MasterDuke17) [inline src/jit/core_templates.expr]: This shouldn't be `(carg $1   ptr)) num_sz))`?

Comment (ugexe) [reply src/jit/core_templates.expr]: > This shouldn't be `(carg $1 ptr)) num_sz))`?

Note this uses the new `calln`, not `call`

File: src/jit/x64/tiles.dasc [hunk 2]
```diff
@@ -827,34 +852,78 @@ MVM_JIT_TILE_DECL(cmp) {
     }
 }
 
+MVM_JIT_TILE_DECL(cmp_num) {
+    MVMint8 left = REG_NUM(tile->values[1]);
+    MVMint8 right = REG_NUM(tile->values[2]);
+    | ucomisd xmm(left), xmm(right);
```
Comment (bdw) [inline src/jit/x64/tiles.dasc]: There's multiple variants of these, and the IR can't express the difference

**Pull request merged**